pkgs/applications/audio/scream/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, fetchFromGitHub, cmake, pkg-config
+, alsaSupport ? true, alsaLib
+, pulseSupport ? false, libpulseaudio
+}:
+
+stdenv.mkDerivation rec {
+  pname = "scream";
+  version = "3.6";
+
+  src = fetchFromGitHub {
+    owner = "duncanthrax";
+    repo = pname;
+    rev = version;
+    sha256 = "01k2zhfb781gfj3apmcjqbm5m05m6pvnh7fb5k81zwvqibai000v";
+  };
+
+  buildInputs = lib.optional pulseSupport libpulseaudio
+    ++ lib.optional alsaSupport alsaLib;
```

Comment:
```suggestion
  buildInputs = lib.optionals pulseSupport [ libpulseaudio ]
    ++ lib.optionals alsaSupport [ alsaLib ];
```

Reply:
This one confuses me, isn't this the exact scenario that `optional` is for (as shown in its example docs)? is `optional` considered bad form or deprecated or something?

Reply:
optional accepts any datatype. optionals only accepts lists and is supposed to be used in conjunction with lists.

Reply:
*points to docs and "flashplayer" example: [lib.lists.optional](https://nixos.org/manual/nixpkgs/stable/#function-library-lib.lists.optional)*

`optional a b` is like a singleton version of optionals, it's equivalent to `optional a (singleton b)` or more simply `optionals a [ b ]`. This is what it's for afaik, when there's only a single item to be added to the list.